prime-router/src/main/kotlin/azure/SettingsFacade.kt

```diff
@@ -275,11 +271,37 @@ class SettingsFacade(
         if (input.organizationName != organizationName) {
             return Triple(false, "Payload and path organization name do not match", null)
         }
-        input.consistencyErrorMessage(metadata)?.let { return Triple(false, it, null) }
         val normalizedJson = JSONB.valueOf(mapper.writeValueAsString(input))
-        return Triple(true, null, normalizedJson)
-    }
+        val errorMessages = when (input) {
+            is OrganizationAPI -> {
+                organizationSchema.validate(mapper.readTree(normalizedJson.toString()))
+            }
+
+            is Sender -> {
+                senderSchema.validate(mapper.readTree(normalizedJson.toString()))
+            }
 
+            is Receiver -> {
+                // validate directly from json parameter to work around a bug where 'mappedConditionFilter'
+                // array of (type: t, value: v) where type got erased
+                receiverSchema.validate(mapper.readTree(normalizedJson.toString()))
+            }
+
+            else -> {
+                throw IllegalStateException("Unexpected input with class = $clazz")
+            }
+        }
+        return if (errorMessages.size > 0) {
+            Triple(false, errorMessages.toString(), normalizedJson)
+        } else {
+            val error = input.consistencyErrorMessage(metadata)
```

Comment:
I think this can get removed?

Reply:
Receiver.consistencyErrorMessage() still doing some mapping schema checking (which is beyond settings yaml schema validation), so would probably keep the function at least for now.